src/olympia/addons/serializers.py

```diff
@@ -1562,23 +1579,28 @@ def fake_object(self, data):
 
         promoted = data.get('promoted', None)
         if promoted:
+            promoted = promoted[0]
             # set .approved_for_groups cached_property because it's used in
             # .approved_applications.
             approved_for_apps = promoted.get('approved_for_apps')
-            obj.promoted = PromotedAddon(
-                addon=obj,
-                approved_application_ids=approved_for_apps,
-                created=None,
-                group_id=promoted['group_id'],
-            )
+            group = PromotedGroup.objects.get(group_id=promoted['group_id'])
```

Comment:
This is the culprit! The extra query we are getting seems to be here where we are explicitly querying the promoted group in the es serializer.